keys/keys.go

```diff
@@ -81,13 +82,18 @@ func ReadPrivateKey(reader io.Reader, passPhrase []byte) (privateKey [chacha20po
 	}
 	if err == nil {
 		// Sometimes the key is returned as a pointer, but sometimes as a value
-		if edPrivateKey, ok := key.(*ed25519.PrivateKey); ok {
-			PrivateKeyToCurve25519(&privateKey, *edPrivateKey)
 
-			return
+		switch v := key.(type) {
+		case ed25519.PrivateKey:
+			edPrivateKey := v
+			PrivateKeyToCurve25519(&privateKey, edPrivateKey)
+		case *ecdh.PrivateKey:
```

Comment:
Symmetry; not pointer to ecdh or why not needed?

Reply:
not sure I follow, can I have more details ?

Reply:
or you mean I should have something like
```
switch v := key.(type) {
		case ed25519.PrivateKey:
			PrivateKeyToCurve25519(&privateKey, v)
		case *ecdh.PrivateKey:
			privateKey = [32]byte(v.Bytes())
		case *ed25519.PrivateKey:
			PrivateKeyToCurve25519(&privateKey, *v)
}
```

Reply:
Sorry if that was unclear, it may be that the case of getting a `ecdh` key as a value never occurs, but then could we document that?

Otherewise, I guess what I would exepect to see would be something like

```
case ecdh.PrivateKey:
    privateKey = v

```

Reply:
ok, added